### PR TITLE
Add io.github.accessory.minus_games_client

### PR DESCRIPTION
**Pull request by Accessory**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
The Minus Games Client purpose is to distribute, run, and sync games hosted by the minus_games_server.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.accessory.minus_games_client.yml [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+      - cargo --offline build --release --verbose --bin minus_games_client
+      - install -Dm755 target/release/minus_games_client /app/bin/minus_games_client
+      - install -Dm644 other/assets/client/io.github.accessory.minus_games_client.svg /app/share/icons/hicolor/scalable/apps/io.github.accessory.minus_games_client.svg
+      - install -Dm644 other/assets/client/io.github.accessory.minus_games_client.desktop /app/share/applications/io.github.accessory.minus_games_client.desktop
+      - install -Dm644 other/assets/client/io.github.accessory.minus_games_client.metainfo.xml /app/share/metainfo/io.github.accessory.minus_games_client.metainfo.xml
+    sources:
+      - type: git
+#        path: ../..
+        url: https://github.com/Accessory/minus_games/
+        branch: main
```
Comment (hfiguiere) [inline io.github.accessory.minus_games_client.yml:166]: NEVER a git branch. this must be a commit and/or tag.

File: io.github.accessory.minus_games_client.yml [hunk 2]
```diff
@@ -0,0 +1,445 @@
...
+      - type: dir
```
Comment (hfiguiere) [inline io.github.accessory.minus_games_client.yml:167]: do not use dir source. also why do you need it?

File: io.github.accessory.minus_games_client.yml [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+        path: .
+      - generated-sources.json
+  - name: start_script
```
Comment (hfiguiere) [inline io.github.accessory.minus_games_client.yml:170]: this should be part of the main module.

File: io.github.accessory.minus_games_client.yml [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+          - export PROTONPATH=GE-Proton
+          - export WINE_EXE=/app/bin/umu-run
+          - export SERVER_URL="http://localhost:8415"
+          - export CLIENT_GAMES_FOLDER="$XDG_DATA_HOME/minus_games_client/games"
+          - export CLIENT_FOLDER="$XDG_DATA_HOME/minus_games_client/client"
+          - export WINE_PREFIX=$XDG_DATA_HOME/minus_games_client/wine/
+          - /app/bin/minus_games_client menu
+
+  # GameMode
+  - name: gamemode
```
Comment (hfiguiere) [inline io.github.accessory.minus_games_client.yml:189]: if this is a dependency it should be before the main app.

File: io.github.accessory.minus_games_client.yml [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+        x-checker-data:
+          type: anitya
+          project-id: 17410
+          stable-only: true
+          url-template: https://github.com/FeralInteractive/gamemode/releases/download/$version/gamemode-$version.tar.xz
+      - type: patch
+        path: patches/379.patch
+
+  # Umu Launcher
+  - name: platform-bootstrap
```
Comment (hfiguiere) [inline io.github.accessory.minus_games_client.yml:210]: same as above.

Comment (hfiguiere) [reply io.github.accessory.minus_games_client.yml:210]: and all the one below.

File: io.github.accessory.minus_games_client.yml (hunk 3 above)
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml]: add to main module

File: io.github.accessory.minus_games_client.yml [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+        commands:
+          - export WINEPREFIX=$XDG_DATA_HOME/wine/
+          - export GAMEID=umu-minus_games_client
+          - export PROTONPATH=GE-Proton
+          - export WINE_EXE=/app/bin/umu-run
+          - export SERVER_URL="http://localhost:8415"
+          - export CLIENT_GAMES_FOLDER="$XDG_DATA_HOME/minus_games_client/games"
+          - export CLIENT_FOLDER="$XDG_DATA_HOME/minus_games_client/client"
+          - export WINE_PREFIX=$XDG_DATA_HOME/minus_games_client/wine/
+          - /app/bin/minus_games_client menu
```
[the remark this answers is not in the document]
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml]: add suggestion

File: io.github.accessory.minus_games_client.yml (hunk 4 above)
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml:150]: main app now at the end

File: io.github.accessory.minus_games_client.yml (hunk 5 above)
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml:171]: main app now at the end

File: io.github.accessory.minus_games_client.yml (hunk 2 above)
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml]: removed

File: io.github.accessory.minus_games_client.yml [hunk 7]
```diff
@@ -0,0 +1,439 @@
...
+  - --filesystem=~/.var/app/com.valvesoftware.Steam:rw
+  - --filesystem=~/.var/app/org.openwinecomponents.umu.umu-launcher:rw
+  - --filesystem=home
+  - --filesystem=host-os
+  - --filesystem=host-etc
+  - --filesystem=xdg-documents
+  - --filesystem=xdg-desktop
+  - --filesystem=xdg-download
+  - --env=TZ=
+  - --unset-env=TZ
```
Comment (hfiguiere) [inline io.github.accessory.minus_games_client.yml:36]: why this?

File: io.github.accessory.minus_games_client.yml [hunk 8]
```diff
@@ -0,0 +1,439 @@
...
+  - --env=LC_ADDRESS=C
+  - --env=LC_COLLATE=C
+  - --env=LC_MONETARY=C
+  - --env=LC_MEASUREMENT=C
+  - --env=LC_NAME=C
+  - --env=LC_NUMERIC=C
+  - --env=LC_TELEPHONE=C
```
Comment (hfiguiere) [inline io.github.accessory.minus_games_client.yml:43]: why overriding the locale.

**Comment by hfiguiere** [comment]
bot, build io.github.accessory.minus_games_client

File: io.github.accessory.minus_games_client.yml [hunk 9]
```diff
@@ -0,0 +1,439 @@
...
+  - --persist=.
+  - --share=ipc
+  - --socket=wayland
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.kde.StatusNotifierWatcher
+  # Required for bwrap to work
+  - --talk-name=org.freedesktop.portal.Background
```
Comment (hfiguiere) [inline io.github.accessory.minus_games_client.yml:67]: are you sure? this is granted by default. And is one of the reason the linter blocked the build.

File: io.github.accessory.minus_games_client.yml [hunk 10]
```diff
@@ -0,0 +1,439 @@
+id: io.github.accessory.minus_games_client
```
Comment (hfiguiere) [inline io.github.accessory.minus_games_client.yml:1]: the check fail because the name doesn't match the repo that is name `minus_games`

File: io.github.accessory.minus_games_client.yml [hunk 11]
```diff
@@ -0,0 +1,439 @@
...
+  - --filesystem=xdg-data/Steam:rw
+  - --filesystem=xdg-data/applications:rw
+  - --filesystem=~/.steam:rw
+  - --filesystem=~/Games:rw
+  - --filesystem=~/.local/share/Steam:rw
+  - --filesystem=~/.var/app/com.valvesoftware.Steam:rw
+  - --filesystem=~/.var/app/org.openwinecomponents.umu.umu-launcher:rw
+  - --filesystem=home
+  - --filesystem=host-os
+  - --filesystem=host-etc
```
Comment (hfiguiere) [inline io.github.accessory.minus_games_client.yml:31]: why are `host-os` and `host-etc` needed?

File: io.github.accessory.minus_games_client.yml [hunk 12]
```diff
@@ -0,0 +1,439 @@
...
+  - --system-talk-name=org.freedesktop.UDisks2
+  # should fix access to SD card on the deck
+  - --filesystem=/run/media
+  # There are still quite a few users using /mnt/ for external drives
+  - --filesystem=/mnt
+  # should fix steamdeck controler navigation
+  - --filesystem=/run/udev:ro
+  # should fix discord rich presence
+  - --filesystem=xdg-run/app/com.discordapp.Discord:create
+  - --persist=.
```
Comment (hfiguiere) [inline io.github.accessory.minus_games_client.yml:58]: This persist is incompatible with `filesystem=home`.

File: io.github.accessory.minus_games_client.yml [hunk 13]
```diff
@@ -0,0 +1,439 @@
...
+  - --allow=bluetooth
+  - --allow=per-app-dev-shm
+  - --env=PATH=/app/bin:/app/utils/bin:/usr/bin:/usr/lib/extensions/vulkan/MangoHud/bin:/usr/lib/extensions/vulkan/gamescope/bin:/usr/lib/extensions/vulkan/OBSVkCapture/bin
+  - --filesystem=xdg-data/Steam:rw
+  - --filesystem=xdg-data/applications:rw
+  - --filesystem=~/.steam:rw
+  - --filesystem=~/Games:rw
+  - --filesystem=~/.local/share/Steam:rw
+  - --filesystem=~/.var/app/com.valvesoftware.Steam:rw
+  - --filesystem=~/.var/app/org.openwinecomponents.umu.umu-launcher:rw
```
Comment (hfiguiere) [inline io.github.accessory.minus_games_client.yml:28]: since you have `filesystem=home` these filesystem permissions are redundant.

File: io.github.accessory.minus_games_client.yml (hunk 7 above)
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml]: removed

File: io.github.accessory.minus_games_client.yml (hunk 8 above)
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml]: removed

File: io.github.accessory.minus_games_client.yml (hunk 9 above)
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml]: removed

File: io.github.accessory.minus_games_client.yml (hunk 10 above)
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml:1]: Created new repo for this

File: io.github.accessory.minus_games_client.yml (hunk 11 above)
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml]: removed

File: io.github.accessory.minus_games_client.yml (hunk 12 above)
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml:36]: removed

File: io.github.accessory.minus_games_client.yml (hunk 13 above)
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml]: removed

File: io.github.accessory.minus_games_client.yml [hunk 14]
```diff
@@ -0,0 +1,411 @@
...
+        ln -srv /app/lib32 /app/links/lib/i386-linux-gnu
+    sources:
+      - type: inline
+        dest-filename: ld.so.conf
+        contents: |
+          # We just make any GL32 extension have higher priority
+          include /run/flatpak/ld.so.conf.d/app-*-org.freedesktop.Platform.GL32.*.conf
+          /app/lib32
+          /app/lib/i386-linux-gnu
+          /lib64
```
Comment (bbhtt) [inline io.github.accessory.minus_games_client.yml:171]: lib64 is not supposed to exist. Where do you see this? It should go to `/lib`.

**Comment by bbhtt** [comment]
bot, build io.github.accessory.minus_games_client

File: flathub.json [hunk 15]
```diff
@@ -0,0 +1,3 @@
+{
+  "only-arches": ["x86_64"]
```
[the remark this answers is not in the document]
Comment (Accessory) [reply flathub.json:2]: For a useful aarch64 use case, changes to the application would need to be made.

File: io.github.accessory.minus_games_client.yml [hunk 16]
```diff
@@ -0,0 +1,411 @@
...
+        sha256: 55af7906a2c75ce6cb280a584776080602444f75815a7aff4d287bb2d7018b32
+  # --- umu ---
+  # We use make first to install things like version and /usr/share/ files
+  # We then use pip to overwrite the python-specific installed files to verify
+  # the symlinks are corrected.
+  - name: umu-run
+    buildsystem: simple
+    build-commands:
+      - |
+        git submodule update --init --recursive
```
[the remark this answers is not in the document]
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml]: Removed

File: io.github.accessory.minus_games_client.yml (hunk 14 above)
Comment (Accessory) [reply io.github.accessory.minus_games_client.yml]: removed

**Comment by bbhtt** [comment]
Can you fix the metainfo? That's the build error

**Comment by bbhtt** [comment]
bot, build io.github.accessory.minus_games_client

**Comment by bbhtt** [comment]
How do I run something? I get "No games found" on all the options.

**Comment by Accessory** [comment]
> How do I run something? I get "No games found" on all the options.

You can find a [Quick Start Guide here](https://github.com/Accessory/minus_games/blob/main/other/docs/quick_start.md)
TLDR: You will probably need to set up a server and connect to it.

**Comment by Accessory** [comment]
I also added a find games option

**Comment by bbhtt** [comment]
Can you link to that quickstart guide in metainfo description?

**Comment by bbhtt** [comment]
Also where are games downloaded from?

**Comment by Accessory** [comment]
> Can you link to that quickstart guide in metainfo description?
Updated the metainfo


> Also where are games downloaded from? "games hosted by the minus_games_server" meaning you upload your own or does it fetch games from somewhere?
ou have to set up a Minus Games Server instance and add games to the instance. However, you can use the Minus Games Client by inserting games into the games folder and then running the games finder from the client's menu.

I do not provide any games or connect to any game downloading service, so you will have to manage your own games.

**Comment by bbhtt** [comment]
OK thanks.

bot, build io.github.accessory.minus_games_client
